### PR TITLE
Add A/B testing info for incremental migration

### PR DESCRIPTION
**Pull request by twsouthwick**

### CODE REVIEW

**Comment by joperezr** [comment]
@Tratcher do we know who is the best person to take a look at this one?

File: aspnetcore/migration/inc/abtesting.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+To enable conditional endpoint selection, a few services need to be defined:
+
+1. Metadata that can be added to an endpoint (controllers, minimal APIs, etc) to turn on any conditional related logic. If this is not present on an endpoint, that endpoint will not incur any potential cost with checks that may be performed.
+
+    ```CSharp
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
+    public sealed class ConditionalRouteAttribute : Attribute
+    {
+        public bool IsConditional { get; set; }
```
Comment (Tratcher) [inline aspnetcore/migration/inc/abtesting.md:29]: Should this default to true? Any routes without this are assumed unconditional.

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
LGTM, but I added Javier for a second look.

**Comment by Rick-Anderson** [comment]
@twsouthwick  I'm going to PR this so you can see and approve what I'm changing. After you approve my PR, it will merge into this branch and we can merge into main.

File: aspnetcore/migration/inc/abtesting.md (hunk 1 above)
Comment (twsouthwick) [reply aspnetcore/migration/inc/abtesting.md]: Yeah, that's a better default. Updated

**Comment by Rick-Anderson** [comment]
@twsouthwick  since you didn't push a branch to this repro but are using a branch from your repo, I can't do a PR into your branch. I'm going to S&M this PR, then I'll do a cleanup PR so you can easily review my changes.

**Pull request merged**